gremlin_client.go

```diff
@@ -116,18 +127,28 @@ func (c *GremlinClient) execWithRetry(ctx context.Context, query string, queryId
 		tryNum++
 
 		if hasKey {
+			fmt.Printf("Attempting to acquire lock for key: %v\n", queryId)
+			acq := time.Now()
 			err = lock.Lock()
 			if err != nil {
 				return nil, err
 			}
+			t := time.Since(acq)
+			fmt.Printf("Acquired lock for key %v after %v\n", queryId, t)
+			queryTimes = append(queryTimes, t)
+			if t > maxTime {
+				maxTime = t
+				fmt.Printf("New longest query: %v\n", maxTime)
+
+			}
+			fmt.Printf("Average query time: %v\n", time.Duration(avgTimeSlice(queryTimes)))
```

Comment:
If we have a logger available it might make sense to use that instead of prints

Reply:
Those were for testing, I'll drop them